sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -207,11 +207,68 @@ private void performLogging(LogLevel logLevel, LogLevel environmentLogLevel, boo
              * Environment is logging at a level higher than verbose, strip out the throwable as it would log its
              * stack trace which is only expected when logging at a verbose level.
              */
-            if (environmentLogLevel.getLogLevel() > LogLevel.VERBOSE.getLogLevel()) {
+            if (rootLogLevel.getLogLevel() > LogLevel.VERBOSE.getLogLevel()) {
                 args = removeThrowable(args);
             }
         }
+        boolean isFromEnv = isLogFromEnv();
+        printInPlace(isFromEnv, throwableMessage, logLevel, format, args);
+
+    }
+
+    private void printInPlace(boolean isFromEnv, String throwableMessage, LogLevel logLevel, String format,
+        Object... args) {
+        if (isFromEnv) {
+            printFromDefaultLogger(throwableMessage, logLevel, format, args);
+        } else {
+            printFromLogger(throwableMessage, logLevel, format, args);
+        }
+    }
+
+    private void printFromDefaultLogger(String throwableMessage, LogLevel logLevel, String format, Object... args) {
+        final DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
```

Comment:
Think more about reducing CPU cycles. This date format does not need to be recreated on every call and could be a static final field in the class.